packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -625,6 +634,17 @@ export function withExtendedResolver(
       getStrictResolver,
     }),
 
+    // These two modules are currently at risk of being duplicated. They're depended on
+    // by various Expo modules, but are mostly included without peer dependencies.
+    // `expo-modules-core` will be replaced by `expo`'s re-exports of it.
+    // `expo-asset` could do the same.
+    // However, neither is present yet
+    // TODO(@kitten): Remove when the above is solved (expo-modules-core and expo-asset are exposed by expo)
+    fallbackModuleResolver.withDependenciesFilter([
+      'expo-asset',
```

Comment:
This wouldn't be needed with the sticky resolver. But without we either:
- add a dependency, and accept it may lead to duplicates
- don't add a dependency, and accept that only the fallback resolver can prevent missing resolution
- re-export it from `expo` like we do with `expo-modules-core` and ensure auto-linking doesn't break